05-services.md

```diff
@@ -784,6 +784,33 @@ duplicates resulting from the merge are not removed.
 
 Any other allowed keys in the service definition should be treated as scalars.
 
+### external
+
+If set to `true`, `external` specifies that this service managed outside of that application defined
+this the Compose file, and might not even rely on containers. A Compose Implementation MUST NOT attempt 
+to manage container(s) for this service. It SHOULD NOT even assume service is actually implemented using 
+containers, and so SHOULD ignore any healthcheck configuration for service, and consider service as "healthy"
+for dependent services.
+
+This attribute is expected to be used in a Compose override file to replace a service definition with an external resource,
+so that compose just ignores this service managing containers for the application, but without the need to fully redefine 
+the application definition.
+
+In the example below, `db` is a database service managed on a cloud infrastructure. 
+
+
+```yaml
+services:
+  app:
+    image: mycompay/app
+    environment:
+      - DB_URL=${DB_URL}
+    depends_on:
+      - db
```

Comment:
in this example, how will compose verify that db is ready before it starts `app`? (the above says that we can't even assume that `db` is a container)

Reply:
with this initial design, it just can't. It's user (or third party tool) responsibility to manage service lifecycle and enforce it's running and healthy.
I wonder how to define some `inspect`, `create` and `delete` logic to be applied to such a service, within the compose specification, offering the ability to run external tools.

Reply:
My assumption here is that the spec hardly can define a portable way to manage lifecycle for a non-container based service (let's say, an AWS RDS database). A Compose Implementation dedicated to this use case could offer extensions to better support this usage, with additional attributes (maybe relying on `x-*` extensions)

Reply:
if other containers can't interact with it, then why put `external: true` in the compose file at all? What problem does it solve then?

Or are you imagining that `external: true` is just there for documentation purposes, and you have to add more fields to it to add behavior?

Reply:
> if other containers can't interact with it,

Obviously they can. They can connect to a RDS database or a FaaS function. But there's no generic way to define lifecycle management for such resources, especially considering compose (at least, Docker Compose) is a client-side implementation which might not have access to the same networks.

> What problem does it solve then? 

My compose file used to declare a database service and I have dependencies defined for this purpose. Then I want to switch to a RDS database. I would need to fully rewrite my compose file to "remove" this db service. With such an `external: true` I can provide an override to just disable service management by compose, and assume some third-party tool ran just before to enforce RDS database is up.

Reply:
Aha! I missed the override use case. Knew I was missing something, thanks!